[IT-1046] Remove `sam publish` from deploy

ConsoleCatzirl opened the pull request:
The lambda function is uploaded to S3 during `sam package` and
we manually copy the built SAM template to S3, so `sam publish`
is not necessary in our use case. Remove it since it is failing
on missing metadata.

zaro0508 commented:
you might want to just remove [sam publish in .travis.yml](https://github.com/Sage-Bionetworks-IT/lambda-cost-reports/blob/master/.travis.yml#L46) instead

ConsoleCatzirl committed:
[IT-1046] Remove `sam publish` from deploy

The lambda function is uploaded to S3 during `sam package` and
we manually copy the built SAM template to S3, so `sam publish`
is not necessary in our use case. Remove it since it is failing
on missing metadata.

ConsoleCatzirl commented:
Ah, I was confused about when the function code was uploaded to S3, but that's done in `sam package` not in `sam publish`. I'll remove the latter since it isn't needed in our use case.
